go-controller/pkg/config/config.go

```diff
@@ -652,6 +653,35 @@ func InitConfigWithPath(ctx *cli.Context, exec kexec.Interface, configFile strin
 	overrideFields(&CNI, &cfg.CNI)
 	overrideFields(&CNI, &cliConfig.CNI)
 
+	// values for token, cacert, kubeconfig, api-server may be found in several places.
+	// Take the first found when looking in this order: command line options, environment variables, files
+
+	if cliConfig.Kubernetes.Token == "" {
+		cliConfig.Kubernetes.Token = os.Getenv("K8S_TOKEN")
+	}
+	if cliConfig.Kubernetes.Token == "" {
+		if data, err1 := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token"); err1 == nil {
+			cliConfig.Kubernetes.Token = string(data)
+		}
+	}
+
+	if cliConfig.Kubernetes.CACert == "" {
+		cliConfig.Kubernetes.CACert = os.Getenv("K8S_CACERT")
+	}
+	if cliConfig.Kubernetes.CACert == "" {
+		if _, err2 := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"); err2 == nil {
```

Comment:
Let's make the /var/run/ stuff constants as well so we avoid possible C&P errors. Just do something like:

```
const (
    kubeServiceAccountFileToken string = "/blahb/albh"
    kubeServiceAccountFileCACert string = "/blahbhbhdhdhd"
)
```

Reply:
Added const(). Moved stuff to buildKubernetesConfig. cli overrides files, overrides this stuff.